proxy.go

```diff
@@ -110,10 +110,40 @@ func main() {
 		Timeout: time.Second * 10,
 	}
 
-	http.HandleFunc(settingsString[LISTEN_PATH], uploadFile)
+	http.HandleFunc("/", proxyHandler) // Hardcoded to / as we need to catch all requests and forward non upload ones
 	http.ListenAndServe(":6743", nil)
 }
 
+func proxyHandler(w http.ResponseWriter, r *http.Request) {
+	if r.Method == http.MethodPost && r.URL.Path == "/api/assets" { // Path is hardcoded as it's the Immich API value, could be made configurable
```

Comment:
Lets not hardcode anything - while this project is currently directly linked with Immich, it's supposed to be compatible with any similiar project  as well.

Reply:
I also think the approach here can be optimized - you are only checking for a POST request, but it might be a PUT request in the future as well. The correct approach would be to detect if its a multipart request with a file attachment, and only trigger the reformat then - otherwise forward to the destination.

Reply:
Nice point, I will change this part to read and reformat only the requests with a multipart/form-data in the API listen_path we define as env variable.